api/proto/teleport/legacy/types/types.proto

```diff
@@ -1580,6 +1582,44 @@ message ProvisionTokenSpecV2GCP {
   repeated Rule Allow = 1 [(gogoproto.jsontag) = "allow,omitempty"];
 }
 
+// ProvisionTokenSpecV2Terraform contains Terraform-specific parts of the
+// ProvisionTokenSpecV2.
+message ProvisionTokenSpecV2Terraform {
+  // Rule is a set of properties the Terraform-issued token might have to be
+  // allowed to use this ProvisionToken.
+  message Rule {
+    // OrganizationID is the ID of the HCP Terraform organization
+    string OrganizationID = 1 [(gogoproto.jsontag) = "organization_id,omitempty"];
+
+    // OrganizationName is the human-readable name of the HCP Terraform organization
+    string OrganizationName = 2 [(gogoproto.jsontag) = "organization_name,omitempty"];
+
+    // ProjectID is the ID of the HCP Terraform project
+    string ProjectID = 3 [(gogoproto.jsontag) = "project_id,omitempty"];
+
+    // ProjectName is the human-readable name for the HCP Terraform project
+    string ProjectName = 4 [(gogoproto.jsontag) = "project_name,omitempty"];
+
+    // WorkspaceID is the ID of the HCP Terraform workspace
+    string WorkspaceID = 5 [(gogoproto.jsontag) = "workspace_id,omitempty"];
+
+    // WorkspaceName is the human-readable name of the HCP Terraform workspace
+    string WorkspaceName = 6 [(gogoproto.jsontag) = "workspace_name,omitempty"];
+
+    // RunPhase is the phase of the run the token was issued for, e.g. `plan` or `apply`
+    string RunPhase = 7 [(gogoproto.jsontag) = "run_phase,omitempty"];
```

Comment:
Should this be a list? Eg, allowing for the `plan` and `apply` phase

Reply:
It can only be one value at a time, and users can specify multiple allowed values using several `allow` rules if they want. Truthfully I'm not even sure whether or not there's a real use case for restricting the run phase, but I figured maybe someone could get creative with it.

An enum would probably be nicer than a string, but annoyingly Terraform doesn't define/document an exhaustive list of values and only provide "plan" and "apply" as examples of possible values :/